data/scripts.js

```diff
@@ -256,18 +256,21 @@ exports.BattleScripts = {
 			if (!targets.length) {
 				this.attrLastMove('[notarget]');
 				this.add('-notarget');
-				return true;
+				return false;
 			}
 			if (targets.length > 1) move.spreadHit = true;
-			damage = 0;
 			let hitTargets = [];
 			for (let i = 0; i < targets.length; i++) {
 				let hitResult = this.tryMoveHit(targets[i], pokemon, move);
 				if (hitResult || hitResult === 0 || hitResult === undefined) {
 					moveResult = true;
 					hitTargets.push(targets[i].toString().substr(0, 3));
 				}
-				damage += hitResult || 0;
+				if (damage !== false) {
+					damage += hitResult || 0;
+				} else {
+					damage = hitResult;
+				}
```

Comment:
Out of interest, what is this doing? I'm interested because it might make it possible to move the "But it failed" message until after all the targets have failed like it does in-game, rather than individually for each target which is how we currently do it. (In particular I think that would need `tryMoveHit` to return `false` when it really failed and `null` in cases where the move was blocked for other reasons.)

Reply:
This is actually so if `hitResult` only returns something other than false one time,  `damage` can retain that value instead of being set to 0, which it always would be before (false + false = 0).